Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,96 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            bool nonExistantCheckWarning = false;
+
+            try
+            {
+                if (externalBoundary.IIsSelfIntersecting())
+                {
+                    Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
+                    return null;
+                }
+            } catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => x.IIsSelfIntersecting()))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is selfintersecting.");
+                    return null;
+                }
+            }
+            catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => !externalBoundary.IIsContaining(x)))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is not contained by the externalBoundary.");
+                    return null;
+                }
+            } catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            bool unionDone = false;
+            try
+            { 
+                for (int i = 0; i < internalBoundaries.Count; i++)
+                {
+                    for (int j = i + 1; j < internalBoundaries.Count; j++)
+                    {
+                        if (internalBoundaries[i].ICurveIntersections(internalBoundaries[j]).Count != 0)
+                        {
+                            internalBoundaries = Compute.BooleanUnion(internalBoundaries).Cast<ICurve>().ToList();
+                            Reflection.Compute.RecordWarning("InternalBounderies were overlapping, BooleanUnion has been used to get non overlapping regions.");
+                            i = internalBoundaries.Count;
+                            j = i;
+                            unionDone = true;
+                        }
+                    }
+                }
+            }
+            catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (!unionDone)
+                {
+                    for (int i = 0; i < internalBoundaries.Count; i++)
```

Comment:
for (int i = 0; i < internalBoundaries.Count - 1; i++)

Reply:
`IsContaining` is not commutative, so the last region must check all the others as well.